lib/t/spotify.ex

```diff
@@ -0,0 +1,40 @@
+defmodule T.Spotify do
+  @moduledoc "spotify api client"
+
+  @spotify_token_url "https://accounts.spotify.com/api/token"
+
+  def token do
+    config = Application.get_env(:t, __MODULE__)
+    token(config)
+  end
+
+  # TODO cache?
+  def token(config) do
+    client_id = Keyword.fetch!(config, :client_id)
+    client_secret = Keyword.fetch!(config, :client_secret)
+
+    auth_key = Base.encode64(client_id <> ":" <> client_secret)
+
+    headers = [
+      {"authorization", "basic " <> auth_key},
+      {"content-type", "application/x-www-form-urlencoded"}
+    ]
+
+    body = "grant_type=client_credentials"
```

Comment:
?

Reply:
required

<img width="835" alt="image" src="https://user-images.githubusercontent.com/16062322/163858895-c359a145-c15f-4a2c-a87d-7bb1eb4ea7d8.png">

from https://developer.spotify.com/documentation/general/guides/authorization/client-credentials/

Reply:
but why in body?

Reply:
I am a novice at this, where should it be?